src/vfShow.ts

```diff
@@ -402,9 +402,43 @@ export class Renderer {
      * @param {Stream} p - a Part or similar object
      */
     prepareTies(p: stream.Stream) {
-        // TODO: bridge voices across measures -- this won't get ties in voices across barlines
-        const p_recursed = <note.GeneralNote[]> Array.from(p.recurse().notesAndRests);
-        // console.log('newSystemsAt', this.systemBreakOffsets);
+        const p_recursed = <note.GeneralNote[]> [];
+        // Determine order for bridging voices: from earliest appearance
+        const voice_ids_in_order_first_encountered = [];
+        // voice IDs are not necessarily unique, so track that they are visited
+        const visited_voices = <stream.Voice[]> [];
+        for (const v of p.recurse().getElementsByClass('Voice')) {
+            if (!voice_ids_in_order_first_encountered.includes(v.id)) {
+                voice_ids_in_order_first_encountered.push(v.id);
+            }
+        }
+        // Retrieve notes in voices
+        for (const v_id of voice_ids_in_order_first_encountered) {
+            for (const v of p.recurse().getElementsByClass('Voice')) {
```

Comment:
note that one way to cast `v` in a for loop is:

```
for (const v of <stream.Voice[]> p.recurse().getElementsByClass('Voice')) {
```

took me a long time to figure that out.  You may need to cast as `<any>` between `<stream.Voice[]>` and `p.recurse()`.